src/Components/Components/src/Rendering/RendererSynchronizationContext.cs

```diff
@@ -47,6 +47,10 @@ public Task Invoke(Action action)
                     action();
                     completion.SetResult(null);
                 }
+                catch (OperationCanceledException)
+                {
+                    completion.SetCanceled();
+                }
```

Comment:
This was a bug in the sync context that I found while unit testing related features. We weren't honouring cancellation and would always return a cancelled task as a faulted task instead.

I unwound some of the test infrastructure that had built up and then found this bug. We should be more careful creating "helpers" instead of writing primitive code.